projects/ad9361/src/ad9361_tinyiiod.c

```diff
@@ -0,0 +1,2721 @@
+ */
+ssize_t set_rf_port_select_available(char *buf, size_t len,
+				     const struct channel_info *channel)
+{
+	return -ENODEV;
+}
+
+/**
+ * set_rf_bandwidth
+ * @param *buff value to be written to attribute
+ * @param len of the value
+ * @param *channel channel properties
+ * @return length of chars written to attribute, or negative value on failure
+ */
+ssize_t set_rf_bandwidth(char *buf, size_t len,
+			 const struct channel_info *channel)
+{
+	ssize_t ret = 0;
+	uint32_t rf_bandwidth = read_ul_value(buf);
+	rf_bandwidth = ad9361_validate_rf_bw(ad9361_phy, rf_bandwidth);
+	if(channel->ch_out) {
+		if(ad9361_phy->current_tx_bw_Hz != rf_bandwidth) {
+			ret = ad9361_update_rf_bandwidth(ad9361_phy, ad9361_phy->current_rx_bw_Hz,
+							 rf_bandwidth);
+		}
+	} else {
+		if(ad9361_phy->current_rx_bw_Hz != rf_bandwidth) {
+			ret = ad9361_update_rf_bandwidth(ad9361_phy, rf_bandwidth,
+							 ad9361_phy->current_tx_bw_Hz);
+		}
+	}
+	if(ret < 0) {
+		return ret;
+	}
+
+	return len;
+}
+
+/**
+ * set_rf_dc_offset_tracking_en
+ * @param *buff value to be written to attribute
+ * @param len of the value
+ * @param *channel channel properties
+ * @return length of chars written to attribute, or negative value on failure
+ */
+ssize_t set_rf_dc_offset_tracking_en(char *buf, size_t len,
+				     const struct channel_info *channel)
+{
+	int8_t en_dis = read_value(buf);
+	if(en_dis < 0) {
+		return en_dis;
+	}
+	ad9361_phy->rfdc_track_en = en_dis ? 1 : 0;
+	if(!channel->ch_out) {
+		return ad9361_tracking_control(ad9361_phy, ad9361_phy->bbdc_track_en,
+					       ad9361_phy->rfdc_track_en, ad9361_phy->quad_track_en);
+	}
+
+	return -ENOENT;
+}
+
+/**
+ * set_sampling_frequency_available
+ * @param *buff value to be written to attribute
+ * @param len of the value
+ * @param *channel channel properties
+ * @return length of chars written to attribute, or negative value on failure
+ */
+ssize_t set_sampling_frequency_available(char *buf, size_t len,
+		const struct channel_info *channel)
+{
+	return -ENODEV;
+}
+
+/**
+ * set_quadrature_tracking_en
+ * @param *buff value to be written to attribute
+ * @param len of the value
+ * @param *channel channel properties
+ * @return length of chars written to attribute, or negative value on failure
+ */
+ssize_t set_quadrature_tracking_en(char *buf, size_t len,
+				   const struct channel_info *channel)
+{
+	int8_t en_dis = read_value(buf);
+	if(en_dis < 0) {
+		return en_dis;
+	}
+	ad9361_phy->quad_track_en = en_dis ? 1 : 0;
+	if(!channel->ch_out) {
+		return ad9361_tracking_control(ad9361_phy, ad9361_phy->bbdc_track_en,
+					       ad9361_phy->rfdc_track_en, ad9361_phy->quad_track_en);
+	}
+
+	return -ENOENT;
+}
+
+/**
+ * set_sampling_frequency
+ * @param *buff value to be written to attribute
+ * @param len of the value
+ * @param *channel channel properties
+ * @return length of chars written to attribute, or negative value on failure
+ */
+ssize_t set_sampling_frequency(char *buf, size_t len,
```

Comment:
'len' is not used in this function, but still returned. Can it be omitted entirely?

Reply:
This set functions, they all have the same signature:
ssize_t (*exec)(char *buf, size_t len, const struct channel_info *channel);
And a reference to the function is used for call.
In some cases "len" parameter is not used, but for a more general implementation, I would leave it there.